docker-compose.yml

```diff
@@ -18,10 +18,3 @@ services:
       - ALLOW_PLAINTEXT_LISTENER=yes
```

Comment:
@StewartJingga Do we need docker-compose anymore for this?

Reply:
for kafka we still need it because we need two containers running, zookeeper and kafka.
And they need to be in the same network (which is being taken care of in docker-compose).

right now dockertest is helpful for spinning up a single container, but since testing kafka requires two containers, I havn't successfully spinned up two containers and connect them in a single network hence I still need docker-compose for this case.

I tried this approach [here](https://github.com/ory/dockertest/issues/195#issuecomment-775895582) mentioned in the dockertest issues but still unable to connect.

Reply:
We can use the new version of Kafka which is without zookeeper.

Reply:
we could but there are two concerns here:
1. The latest official kafka docker image is 2.8.0 which is also the same version where KRaft (Kafka Raft mode to replace zookeeper) is introduced. They put a warning [here](https://github.com/apache/kafka/blob/2.8/config/kraft/README.md#warning) that it is still for testing and would be production ready on `v3.0`.
2. We would need to create our own docker image to enable KRaft mode since a few configurations have not been exposed by the latest docker image and need to be tweaked manually.

A few users have created a kafka image with KRaft mode enabled which we can directly use, but I'm not sure it is a good idea to use that image since mostly it is for their use cases and we can't guarantee stability.

Reply:
Since it is for integration tests. I think it should be fine. Later we can move it to 3.0.

Reply:
ok sure, then we need create our own docker image to enable KRaft mode.

Do we have a repo to store our Dockerfiles?

Reply:
is there any public build image with Kraft that we can use?

Reply:
Maybe we could put the dockerfile in the meteor repo for now since atm it is only being used by its kafka extractor.
e.g.
```
plugins/extractors/kafka/Dockerfile
```

wdyt?

Reply:
> is there any public build image with Kraft that we can use? if yes, let's use that. If not then it is ok for us to wait till it is released.

found one [here](https://www.morling.dev/blog/exploring-zookeeper-less-kafka/).

Reply:
oh wait, even that one is not yet published to dockerhub, we still need to build the image ourselves 😞 .

Found another one here in dockerhub https://hub.docker.com/r/moeenz/docker-kafka-kraft.
It has `141 Downloads` and `0 stars` as of now. It probably works fine, I can try it, but are you positive with using unverified public image?

Reply:
Since this is only for integration testing. it should be fine. We will move to a more stable image when released.